frontend/src/metabase/core/components/TabButton/TabButton.unit.spec.tsx

```diff
@@ -26,14 +26,15 @@ function setup(props?: Partial<RenameableTabButtonProps>) {
       />
     </TabRow>,
   );
+
   return { action, onRename, value };
 }
 
 describe("TabButton", () => {
   it("should open the menu upon clicking the chevron", async () => {
     setup();
 
-    await userEvent.click(getIcon("chevrondown"));
+    await act(async () => await userEvent.click(getIcon("chevrondown")));
```

Comment:
Could you please tell why do we need to wrap `userEvent` usages with `act` here?

Reply:
It's mainly because it looks cleaner when running the tests in a console (i.e. no warnings). Theoretically, wrapping user interactions in act() makes it so any incidental asynchronous operation is completed before moving to the next statement in the test, ensuring the test's replicability. Here, I don't think it really makes a difference because we're using `findBy` later on anyway, which is itself waiting for the UI to stabilize.

TL;DR: warnings don't look good in the console :-)

Reply:
Okay I re-read some doc about this, it turns out I was a bit too enthusiastic with using `act()`. It's cleaner now.